### ISSUE TITLE
Баг с итем сета мага

### ISSUE THREAD

**Issue by Ramusik** [tobmaps/TrinityCore#98]
Вобщем есть следующий баг:
1. Взять магу 2Т10 - http://db.uawow.com/?spell=67164
2. Наложить на себя любой из выше перечисленных щитов, например http://db.uawow.com/?spell=43046
3. Поменять шмотки на любые другие, например на т10
4. Бонус от 2 частей остается на примененном доспехе

Т.е. получается, снятие ауры бонуса 2Р не оказывает влияния на уже активную ауру описанных доспехов в бонусе. Таким образом, файры неплохо багают себе крит, а арканы - манареген.

Пока что поставил себе заплатку:
http://paste2.org/p/1407316

Но надеюсь поможешь исправить более целесообразно, так как я думаю есть похожие баги и с другими схожими талантами\спелами

**Comment by tobmaps** [comment]
Вообще подобных проблем просто навалом, похоже что нужно как-то очищать триггерные спеллы при снятии ауры их запустившей. Выходит нужно хранить некий список для каждой ауры, который будет содержать список стриггеренных ею аур. В общем с этим пока всё глухо

**Comment by Ramusik** [comment]
Может есть какое то более общее решение для снятия щитов магов ? так как они могут багать аналогичным способом Раскаленный щит еще и с помощью http://www.wowhead.com/item=42751... Хотя бы хаково, чтобы при снятии эффектов http://db.uawow.com/?spell=67164 и http://db.uawow.com/?spell=56382 снимались все ауры щитов с мага

**Comment by tobmaps** [comment]
Я узнал что на оффе ПРИ СБРОСЕ/ПЕРЕКЛЮЧЕНИИ ТАЛОВ все щиты и т.д. полезные бафы с тебя снимаются, пока не узнал только - снимаются ли бафы которые улучшены спеллмодами и т.д., или же всегда и все. Слышал что например [Печать праведности] у человека не снималась, когда баф [Благославление могущества] сбрасывался. Причем снимаются только собственные бафы.
Также снимается и стелс

Найти бы человека с оффа который бы все эти вопросы разрулил

Или подгоните мне аккаунт xD

**Comment by Ramusik** [comment]
Провел небольшое исследование и вот что откопал - AttributesEx7: 0x10000000 (SPELL_ATTR7_UNK28) как раз имеют ауры, такие как доспехи магов, благословения, шипы, стойкость - т.е. как раз те позитивные бафы, которые должны сниматься с игрока при смене спека / сброса талантов)

**Comment by Ramusik** [comment]
Насчет аттрибутов - вот еще

SPELL_ATTR7_UNK8  ==>> SPELL_ATTR7_HORDE_ONLY
SPELL_ATTR7_UNK9  ==>> SPELL_ATTR7_ALLIANCE_ONLY

Мне кажеться спеллы с этими аттрибутами не должны быть доступны для использования противоположным фракциям.

**Comment by tobmaps** [comment]
Находка хорошая

**Comment by tobmaps** [comment]
Нужно узнать что происходит со спеллами Порталов в столицы при рероле с орды в альянс и наоборот, мб это отвечает за Обмен спеллов при рероле

**Comment by Ramusik** [comment]
Нет, Bracers of Swift Death 67139 - это ордынские, точно также как и Жажда Крови (SPELL_ATTR7_UNK8 )

А у альянса - Bracers of Swift Death - 67087

**Comment by Ramusik** [comment]
> Нужно узнать что происходит со спеллами Порталов в столицы при рероле с орды в альянс и наоборот, мб это отвечает за Обмен спеллов при рероле

Они просто заменяются на версию противоположной фракции. Сейчас это реализовано через базу ...

**Comment by Ramusik** [comment]
Написал решение проблемы, когда при сбросе талантов / смене спека не сбрасывались специальные ауры (доспехи, благословения и т.д.)

http://paste2.org/p/1421120

Осталась только проблема со сбросом доспехов при снятии Т9 )))

**Comment by tobmaps** [comment]
ага и 

```
ID - 6562 Heroic Presence (Racial Passive)
ID - 28878 Heroic Presence (Racial Passive) 
```

снимет

а

```
ID - 63611 Improved Blood Presence
```

и все остальные "власти" у нас всё ещё можно забажить сбросом талов

**Comment by Ramusik** [comment]
Насчет расовой дренеев че то забыл - тогда нужно добавить проверку на пассивные ауры:

http://paste2.org/p/1421137

**Comment by tobmaps** [comment]
По-моему этот аттрибут больше отвечает за стак аур нежели за резет талов. Хотя...

**Comment by Ramusik** [comment]
Ну другого решения пока не видно (( По крайней мере вроде больше не видно общих аттрибутов у этих спеллов

**Comment by Ramusik** [comment]
В принципе может добавишь в свой форк патч? Заодно народ протестирует, если что не так - отпишется
